core/src/main/java/com/cyberbot/bomberman/core/models/tiles/loader/TileMapFactory.java

```diff
@@ -0,0 +1,120 @@
+package com.cyberbot.bomberman.core.models.tiles.loader;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.physics.box2d.World;
+import com.cyberbot.bomberman.core.models.tiles.*;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.File;
+import java.io.IOException;
+import java.util.InvalidPropertiesFormatException;
+
+public class TileMapFactory {
+    private static final String TAG = TileMapFactory.class.getSimpleName();
+
+    private static final String LAYER_BASE = "base";
+    private static final String LAYER_FLOOR = "floor";
+    private static final String LAYER_WALLS = "walls";
+
+
+    public static TileMap createTileMap(World world, String path) throws IOException, MissingLayersException, ParserConfigurationException, SAXException {
+
+        Node sourceMap = loadXmlMap(path);
+        Element mapElement = (Element) sourceMap;
+        NodeList tilesetSource = mapElement.getElementsByTagName("tileset");
+        Element tilesetSourceElement = (Element) tilesetSource.item(0);
+
+        NodeList tileset = loadXmlTileset("./map/" + tilesetSourceElement.getAttribute("source"));
+
+        TileMapLayer baseLayer = null;
+        TileMapLayer floorLayer = null;
+        TileMapLayer wallsLayer = null;
+
+
+        NodeList layers = mapElement.getElementsByTagName("layer");
+
+        for (int i = 0; i < layers.getLength(); i++) {
+            Element element = (Element) layers.item(i);
+
+            Tile[][] tiles = csvStringToTileArray(element.getElementsByTagName("data").item(0).getTextContent(),
+                Integer.parseInt(element.getAttribute("width")),
+                Integer.parseInt(element.getAttribute("height")),
+                tileset,
+                world);
+
+            TileMapLayer mapLayer = new TileMapLayer(Integer.parseInt(element.getAttribute("width")),
+                Integer.parseInt(element.getAttribute("height")), tiles);
+            switch (element.getAttribute("name")) {
+                case LAYER_BASE:
+                    baseLayer = mapLayer;
+                    break;
+                case LAYER_FLOOR:
+                    floorLayer = mapLayer;
+                    break;
+                case LAYER_WALLS:
+                    wallsLayer = mapLayer;
+                    break;
+                default:
+                    Gdx.app.error(TAG, "Unsupported tile layer found: '" +
+                        element.getAttribute("name") + "', will be ignored");
+            }
+        }
+        return new TileMap(baseLayer, floorLayer, wallsLayer);
+    }
+
+    private static Node loadXmlMap(String path) throws IOException, ParserConfigurationException, SAXException {
+
+        File inputFile = new File(path);
```

Comment:
Better to pass `File` as the argument here.

```suggestion
    private static Node loadXmlMap(File file) throws IOException, ParserConfigurationException, SAXException {

        // Rename inputFile to file
        // File inputFile = new File(path);
```

Reply:
Fixed